src/main/java/graphql/language/AstPrinter.java

```diff
@@ -184,8 +184,8 @@ private AstPrinter(boolean compactMode) {
         };
     }
 
-    private boolean hasComments(List<? extends Node> nodes) {
-        return nodes.stream().anyMatch(it -> it.getComments().size() > 0);
+    private boolean hasDescription(List<? extends Node> nodes) {
+        return nodes.stream().anyMatch(it -> ((AbstractDescribedNode) it).getDescription() != null);
```

Comment:
safety might be to filter first

`stream().filter( it -> it instanceof AbstractDescribedNode)..anyMatch(it -> ((AbstractDescribedNode) it).getDescription() != null)`